rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -154,941 +218,203 @@ impl Handler<GetPhyIpFromMeshIp> for TunnelManager {
     }
 }
 
-pub struct UnListen(pub String);
-impl Message for UnListen {
-    type Result = ();
-}
-
-impl Handler<UnListen> for TunnelManager {
-    type Result = ();
-
-    fn handle(&mut self, un_listen: UnListen, _: &mut Context<Self>) -> Self::Result {
-        self.listen_interfaces.remove(&un_listen.0);
-        SETTING.get_network_mut().peer_interfaces = self.listen_interfaces.clone();
-    }
-}
-
-pub struct GetListen;
-impl Message for GetListen {
-    type Result = Result<HashSet<String>, Error>;
-}
-
-impl Handler<GetListen> for TunnelManager {
-    type Result = Result<HashSet<String>, Error>;
-    fn handle(&mut self, _: GetListen, _: &mut Context<Self>) -> Self::Result {
-        Ok(self.listen_interfaces.clone())
-    }
-}
-
 pub struct GetNeighbors;
 impl Message for GetNeighbors {
     type Result = Result<Vec<(LocalIdentity, String, IpAddr)>, Error>;
 }
 
 impl Handler<GetNeighbors> for TunnelManager {
-    type Result = ResponseFuture<Vec<(LocalIdentity, String, IpAddr)>, Error>;
+    type Result = Result<Vec<(LocalIdentity, String, IpAddr)>, Error>;
 
     fn handle(&mut self, _: GetNeighbors, _: &mut Context<Self>) -> Self::Result {
-        self.get_neighbors()
-    }
-}
-
-pub struct GetLocalIdentity {
-    pub from: IpAddr,
-}
-impl Message for GetLocalIdentity {
-    type Result = LocalIdentity;
-}
-
-impl Handler<GetLocalIdentity> for TunnelManager {
-    type Result = MessageResult<GetLocalIdentity>;
-
-    fn handle(&mut self, their_id: GetLocalIdentity, _: &mut Context<Self>) -> Self::Result {
-        MessageResult(self.get_local_identity(their_id.from))
+        let mut res = Vec::new();
+        for tunnel in self.tunnels.iter() {
+            res.push((tunnel.localid.clone(), tunnel.iface_name.clone(), tunnel.ip));
```

Comment:
Might wanna `map().collect()` here instead, which could make the handler leaner. I understand the paing of tracking long method chains, but that's not really the case here.